packages/extension-browser/src/devtools/views/app.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import { Config as ConfigData, Results as ResultsData } from '../../shared/types';
+
+import AnalyzePage from './pages/analyze';
+import ConfigPage from './pages/config';
+import ResultsPage from './pages/results';
+
+import { trackCancel, trackFinish, trackStart } from '../utils/analytics';
+import { sendMessage } from '../utils/messaging';
+import { addNetworkListeners, removeNetworkListeners } from '../utils/network';
+import { useCurrentTheme } from '../utils/themes';
+
+import * as styles from './app.css';
+
+const enum Page {
+    Analyze,
+    Config,
+    Results
+}
+
+const emptyResults: ResultsData = { categories: [] };
+
+const App = () => {
+    const [page, setPage] = useState(Page.Config);
+    const [results, setResults] = useState(emptyResults);
+    const theme = useCurrentTheme();
+
+    const onCancel = (duration: number) => {
+        setPage(Page.Config);
+        sendMessage({ done: true }); // Tell the background script to cancel scanning.
+        removeNetworkListeners();
+        trackCancel(duration);
+    };
+
+    const onRestart = () => {
+        setPage(Page.Config);
+    };
+
+    const onResults = (results: ResultsData, duration: number) => {
+        setPage(Page.Results);
+        setResults(results);
+        trackFinish(duration);
+    };
+
+    const onStart = (config: ConfigData) => {
+        setPage(Page.Analyze);
+        sendMessage({ enable: { config } }); // Tell the background script to start scanning.
```

Comment:
Maybe we should document the messages that can be sent somewhere?

Reply:
All the possible messages are defined in `src/shared/types.ts`, but I'm happy to add additional comments there explaining what each is used for. 

That said, I've been debating refactoring this a bit to have a single utils file with methods to call for each action that sends messages (and only have that file actually call `sendMessage`).

Reply:
> That said, I've been debating refactoring this a bit to have a single utils file with methods to call for each action that sends messages (and only have that file actually call `sendMessage`).

Open an issue for the future and call it a day?